server/controllers/LibraryController.js

```diff
@@ -218,30 +252,55 @@ class LibraryController {
           }
         })
       }
-      libraryItems = naturalSort(libraryItems).by(sortArray)
     }
 
-    if (payload.collapseseries) {
-      libraryItems = libraryHelpers.collapseBookSeries(libraryItems, this.db.series)
-      payload.total = libraryItems.length
-    } else if (filterSeries) {
-      // Book media when filtering series will include series object on media metadata
-      libraryItems = libraryItems.map(li => {
-        var series = li.media.metadata.getSeries(filterSeries)
-        var liJson = payload.minified ? li.toJSONMinified() : li.toJSON()
-        liJson.media.metadata.series = series
-        return liJson
-      })
-      libraryItems = naturalSort(libraryItems).asc(li => li.media.metadata.series.sequence)
-    } else {
-      libraryItems = libraryItems.map(li => payload.minified ? li.toJSONMinified() : li.toJSON())
+    if (sortArray.length) {
+      libraryItems = naturalSort(libraryItems).by(sortArray)
     }
 
-    if (payload.limit) {
-      var startIndex = payload.page * payload.limit
-      libraryItems = libraryItems.slice(startIndex, startIndex + payload.limit)
-    }
-    payload.results = libraryItems
+    // Step 4 - Transform the items to pass to the client side
+    payload.results = libraryItems.map(li => {
+      let json = payload.minified ? li.toJSONMinified() : li.toJSON()
+
+      if (li.collapsedSeries) {
+        json.collapsedSeries = {
+          id: li.collapsedSeries.id,
+          name: li.collapsedSeries.name,
+          nameIgnorePrefix: li.collapsedSeries.nameIgnorePrefix,
+          libraryItemIds: li.collapsedSeries.books.map(b => b.id),
+          numBooks: li.collapsedSeries.books.length
+        }
+
+        // If collapsing by series and filtering by a series, generate the list of sequences the collapsed
+        // series represents in the filtered series
+        if (filterSeries) {
+          json.collapsedSeries.seriesSequenceList = li.collapsedSeries.books
+            .map(b => parseFloat(b.filterSeriesSequence))
```

Comment:
An issue with this solution to overflow sequence values is that we don't enforce a sequence to be a number. For example, a valid sequence could be roman numerals or `1a`. In that case this would crash but if we chose to just ignore those then the possibility of the sequence overflowing the book cover would remain.

Reply:
I fixed the logic to use a natural sort. Even without that change, it still wouldn't overflow. The overflow before was caused by a lack of whitespace. I implemented ranges because I think they are more readable, not because they fixed overflow. Below you can see how it looks. I also updated it to use the same styling as normal sequence numbers (I think it looks better).

![Screenshot_20221104_090103](https://user-images.githubusercontent.com/2041129/200104071-3e5927f7-3201-490d-b193-141e087384ce.png)